options.go

```diff
@@ -77,6 +84,19 @@ func checkOptions(rp *Ringpop) []error {
 
 // Runtime options
 
+// Clock is used to set the Clock mechanism. Passing nil will use the system
+// clock. Testing harnesses will typically replace this with a mocked clock.
+func Clock(c clock.Clock) Option {
+	return func(r *Ringpop) error {
+		if c == nil {
```

Comment:
They are using different options mechanisms, though.

The functional option gives us precisely this advantage. We can differentiate between `nil` and not having anything passed. There is no reason why we would ever want to accept a nil clock, so I think doing so explicitly should be an error.

If you want the default clock, you don't need to specify this option at all.

For the swim node, it's not as relevant, since the package is internal (well, it's meant to be and/or will be as soon as we get around to it) and I think that if we're taking in a Clock from the user from Ringpop we need to pass that on in swim as well. So we're always going to be passing the Ringpop clock.